site_scons/site_tools/compiler_setup.py

```diff
@@ -51,6 +51,12 @@ def _base_setup(env):
                         '-Wall',
                         '-fpic'])
 
+    if env["ADDRESS_SANITIZER"] == 1:
+        print('Enabling Address Sanitizer')
+        # Enable Address Sanitizer, a fast memory error detector.
+        env.Prepend(LINKFLAGS=['-fsanitize=address'])
+        env.Prepend(CCFLAGS=['-fsanitize=address'])
```

Comment:
Also `'-fno-omit-frame-pointer'` in `CCFLAGS` for better reports.

I found that I had to change `-Wframe-larger-than=4096` to a larger value (8192 worked for me), due to the redzones added by asan.